observability/observability.go

```diff
@@ -0,0 +1,88 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package observability implements the tracing, metrics, and logging data
+// collection, and provides controlling knobs via a config file.
+//
+// Experimental
+//
+// Notice: This API is EXPERIMENTAL and may be changed or removed in a
+// later release.
+package observability
+
+import (
+	"context"
+
+	"google.golang.org/grpc/binarylog"
+	"google.golang.org/grpc/grpclog"
+	configpb "google.golang.org/grpc/observability/internal/config"
+)
+
+var (
+	logger = grpclog.Component("observability")
+	config *configpb.ObservabilityConfig
```

Comment:
Why is this a global?  Can it _not_ be?

Reply:
There are 2 places need config:

1. In the new `Init` function, using the config to setup binary logging;
2. In the `Start` function, using the config to setup exporters.

I used a global variable to ensure we only parse once. This ensures the binary logging won't be invalidated by other `Init` changing the config in someway. Alternatively, I can minimize the global variable to binary logging control string, and throw an error if the config in `Start` doesn't match `Init`. WDYT?